### ISSUE TITLE
Evironment not being passed through to the xUnit process

### ISSUE THREAD

**Issue by gbrackley** [carlpett/xUnit-TeamCity#5]
Great plugin thank you. When running tests we are not seeing the teamcity environment being passed through to the xUnit process.  This means our tests fail becuase they get configed via environment variables (based on which deployment environment they should be testing against).

I wrote the following test to dump the environment variables when a test runs (on a teamcity build agent).

```
        [Fact]
        public void DumpEnvironment()
        {
            var env = System.Environment.GetEnvironmentVariables();

            Log.InfoFormat("{0} environment variables", env.Count);
            foreach (var key in env.Keys)
            {
                Log.InfoFormat("{0}={1}", key, env[key]);
            }


            env.ShouldNotBeNull();
            env.ShouldNotBeEmpty();
        }
```

My expectation is that I would see both teamcity environment variables _and_ our test specific configuration variables. What I am seeing is standard system environment variables only.

What we see is the following output (the width has been truncated for clarity):

```
DumpEnvironmentTest.DumpEnvironment - 37 environment variables
DumpEnvironmentTest.DumpEnvironment - COMPUTERNAME=WIN-1637E2HD7UB
DumpEnvironmentTest.DumpEnvironment - VS140COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Commo
DumpEnvironmentTest.DumpEnvironment - PUBLIC=C:\Users\Public
DumpEnvironmentTest.DumpEnvironment - LOCALAPPDATA=C:\Windows\system32\config\systemprofile\AppData\Local
DumpEnvironmentTest.DumpEnvironment - PSModulePath=C:\Windows\system32\WindowsPowerShell\v1.0\Modules\;C:\Prog
DumpEnvironmentTest.DumpEnvironment - PROCESSOR_ARCHITECTURE=AMD64
DumpEnvironmentTest.DumpEnvironment - Path=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\
DumpEnvironmentTest.DumpEnvironment - CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
DumpEnvironmentTest.DumpEnvironment - ProgramFiles(x86)=C:\Program Files (x86)
DumpEnvironmentTest.DumpEnvironment - PROCESSOR_LEVEL=6
DumpEnvironmentTest.DumpEnvironment - ProgramFiles=C:\Program Files
DumpEnvironmentTest.DumpEnvironment - PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
DumpEnvironmentTest.DumpEnvironment - USERPROFILE=C:\Windows\system32\config\systemprofile
DumpEnvironmentTest.DumpEnvironment - SystemRoot=C:\Windows
DumpEnvironmentTest.DumpEnvironment - ALLUSERSPROFILE=C:\ProgramData
DumpEnvironmentTest.DumpEnvironment - FP_NO_HOST_CHECK=NO
DumpEnvironmentTest.DumpEnvironment - ProgramData=C:\ProgramData
DumpEnvironmentTest.DumpEnvironment - PROCESSOR_REVISION=3f02
DumpEnvironmentTest.DumpEnvironment - USERNAME=WIN-1637E2HD7UB$
DumpEnvironmentTest.DumpEnvironment - CommonProgramW6432=C:\Program Files\Common Files
DumpEnvironmentTest.DumpEnvironment - WRAPPER_OS=windows
DumpEnvironmentTest.DumpEnvironment - CommonProgramFiles=C:\Program Files\Common Files
DumpEnvironmentTest.DumpEnvironment - OS=Windows_NT
DumpEnvironmentTest.DumpEnvironment - WRAPPER_PATH_SEPARATOR=;
DumpEnvironmentTest.DumpEnvironment - PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 63 Stepping 2, GenuineIntel
DumpEnvironmentTest.DumpEnvironment - ComSpec=C:\Windows\system32\cmd.exe
DumpEnvironmentTest.DumpEnvironment - WRAPPER_ARCH=x86
DumpEnvironmentTest.DumpEnvironment - SystemDrive=C:
DumpEnvironmentTest.DumpEnvironment - TEMP=C:\Windows\TEMP
DumpEnvironmentTest.DumpEnvironment - WRAPPER_BITS=32
DumpEnvironmentTest.DumpEnvironment - NUMBER_OF_PROCESSORS=2
DumpEnvironmentTest.DumpEnvironment - APPDATA=C:\Windows\system32\config\systemprofile\AppData\Roaming
DumpEnvironmentTest.DumpEnvironment - TMP=C:\Windows\TEMP
DumpEnvironmentTest.DumpEnvironment - ProgramW6432=C:\Program Files
DumpEnvironmentTest.DumpEnvironment - windir=C:\Windows
DumpEnvironmentTest.DumpEnvironment - USERDOMAIN=WORKGROUP
DumpEnvironmentTest.DumpEnvironment - WRAPPER_FILE_SEPARATOR=\
```

I also added a teamcity script build task that did a "set > filename" as a way to capture the environment in a standard task. The output had lots of environment variables:

```
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Windows\system32\config\systemprofile\AppData\Roaming
BUILD_NUMBER=95
BUILD_VCS_NUMBER=6f1b017aa4bdb5b3115ea206d02e2e02f61b0265
BUILD_VCS_NUMBER_cx=6f1b017aa4bdb5b3115ea206d02e2e02f61b0265
CemApiUri=https://api-cem-dev.example.com/
CommonProgramFiles=C:\Program Files (x86)\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=WIN-1637E2HD7UB
ComSpec=C:\Windows\system32\cmd.exe
FP_NO_HOST_CHECK=NO
LOCALAPPDATA=C:\Windows\system32\config\systemprofile\AppData\Local
NUMBER_OF_PROCESSORS=2
OS=Windows_NT
Path=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\ManagementStudio\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Windows\system32\config\systemprofile\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
PROCESSOR_ARCHITECTURE=x86
PROCESSOR_ARCHITEW6432=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 63 Stepping 2, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=3f02
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files (x86)
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\Windows\system32\WindowsPowerShell\v1.0\Modules\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\PowerShell\Modules\
PUBLIC=C:\Users\Public
SystemDrive=C:
SystemRoot=C:\Windows
TEAMCITY_BUILDCONF_NAME=Acceptance Test
TEAMCITY_BUILD_PROPERTIES_FILE=C:\BuildAgent\temp\buildTmp\teamcity.build8232542898749439084.properties
TEAMCITY_CAPTURE_ENV="c:\BuildAgent\jre\bin\java.exe" -jar "C:\BuildAgent\plugins\environment-fetcher\bin\env-fetcher.jar"
TEAMCITY_GIT_PATH=C:\Program Files\Git\bin\git.exe
TEAMCITY_JRE=C:\BuildAgent\jre
TEAMCITY_PROCESS_FLOW_ID=14868242813647
TEAMCITY_PROCESS_PARENT_FLOW_ID=
TEAMCITY_PROJECT_NAME=CX
TEAMCITY_VERSION=9.1.4 (build 37293)
TEMP=C:\BuildAgent\temp\buildTmp
TMP=C:\BuildAgent\temp\buildTmp
TMPDIR=C:\BuildAgent\temp\buildTmp
USERDOMAIN=WORKGROUP
USERNAME=WIN-1637E2HD7UB$
USERPROFILE=C:\Windows\system32\config\systemprofile
VS140COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\
windir=C:\Windows
WRAPPER_ARCH=x86
WRAPPER_BITS=32
WRAPPER_FILE_SEPARATOR=\
WRAPPER_OS=windows
WRAPPER_PATH_SEPARATOR=;
```

I did a quick check of the code but didn't immediately see an obvious way to prove the plugin was causing the issue or how to fix it. Is this an issue with the plugin or is it the xunit console runner?

**Comment by carlpett** [comment]
Hi,
Thanks for the bug report! The environment _should_ be passed along to the spawned process according to the java `ProcessBuilder` docs, but I'll try to verify this actually works. There might be some hoop-hopping since it is running within TeamCity.

**Comment by JustinMorgan** [comment]
I've been experiencing this same issue and wondering if I was going crazy. I did a similar dump of the environment during the test run and got the exact same result: plenty of environment variables, but nothing that came from TeamCity. I think it must be an issue with the plugin--or with TeamCity itself--because the xunit console runner shouldn't be able to tell the difference between environment variables from TC and the ones already on the machine.

**Comment by carlpett** [comment]
Yeah, it's not in the xunit runner. It seems the teamcity agent might not initialize it's own environment as I had expected. 
"Life happened" a bit last week, so I've been a bit tied up lately and unable to focus on this. I hope to have some progress soon, though.

**Comment by JustinMorgan** [comment]
No sweat, thanks for the update. Other than this, the plugin has been really easy to use so far. I appreciate you putting it out there.

**Comment by carlpett** [comment]
@JustinMorgan This is fixed in 1.1. I didn't know GitHub automatically closed issues when mentioned in a commit, and there is still just a prerelease. Sorry about that.
